wade-review.md

```diff
@@ -0,0 +1,596 @@
+                        GitHubPRStateCounter entry = kvStore.get(STORE_KEY);
+                        System.out.printf("Store value %s%n", entry);
+                        context.forward(new Record<>("pr", entry, timestamp));
+                    });
+	}
+}
+
+```
+
+Then, I’ll initialize it with our state store. The init method schedules a punctuation to fire every second, printing the ratio in the store. 
+
+```java
+static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> { 
+	static final String STORE_KEY = "state-counter";
+   @Override
+   public Processor<String, GitHubPRInfo, String, GitHubPRStateCounter> get() {
+		return new Processor<>() {
+
+    private KeyValueStore<String, GitHubPRStateCounter> kvStore;
+                @Override
+                public void init(final ProcessorContext<String, GitHubPRStateCounter> context) {
+                    this.kvStore = context.getStateStore(STORE_NAME);
+
+                    context.schedule(Duration.ofSeconds(1), PunctuationType.STREAM_TIME, timestamp -> {
+                        GitHubPRStateCounter entry = kvStore.get(STORE_KEY);
+                        System.out.printf("Store value %s%n", entry);
+                        context.forward(new Record<>("pr", entry, timestamp));
+                    });
+	}
+              @Override
+                public void process(final Record<String, GitHubPRInfo> record) {
+
+                }
+            };
+        }
+}
+```
+
+Next, I’ll write the process method to  take in the events, mark them as open or closed, increment the count, and stash them in the state store.
+
+```java
+static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> { 
+	static final String STORE_KEY = "state-counter";
+   @Override
+   public Processor<String, GitHubPRInfo, String, GitHubPRStateCounter> get() {
+		return new Processor<>() {
+
+    private KeyValueStore<String, GitHubPRStateCounter> kvStore;
+                @Override
+                public void init(final ProcessorContext<String, GitHubPRStateCounter> context) {
+                    this.kvStore = context.getStateStore(STORE_NAME);
+
+                    context.schedule(Duration.ofSeconds(1), PunctuationType.STREAM_TIME, timestamp -> {
+                        GitHubPRStateCounter entry = kvStore.get(STORE_KEY);
+                        System.out.printf("Store value %s%n", entry);
+                        context.forward(new Record<>("pr", entry, timestamp));
+                    });
+	}
+              @Override
+                public void process(final Record<String, GitHubPRInfo> record) {
+                    GitHubPRInfo prInfo = record.value();
+
+                }
+            };
+        }
+}
+```
+We’ll get the value of the record and start a chain of logic. 
+
+```java
+SHOULD I ZOOM IN HERE I WONDER
+                    if(!prInfo.state().equals(NO_STATE)) {
```

Comment:
I think zooming is ok. However, I would leave the `prInfo = record.value()` line on the screen. Basically just zoom in on the internals of the method.

Reply:
https://github.com/Cerchie/notes-for-gh-video/pull/2/commits/ea2dbc52edc320cfae68f0c07a94ffac6c875577